src/main/java/com/weather/assignment/handler/WeatherHandler.java

```diff
@@ -0,0 +1,68 @@
+package com.weather.assignment.handler;
+
+import com.weather.assignment.document.WeatherElements;
+import com.weather.assignment.repository.WeatherRepository;
+import com.weather.assignment.service.WeatherService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.BodyInserters;
+import org.springframework.web.reactive.function.server.ServerRequest;
+import org.springframework.web.reactive.function.server.ServerResponse;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.Date;
+
+@Component
+public class WeatherHandler {
+
+    @Autowired
+    WeatherRepository weatherRepository;
```

Comment:
- The fields should always be `private` unless it is absolutely necessary.
- Always use constructor injection.